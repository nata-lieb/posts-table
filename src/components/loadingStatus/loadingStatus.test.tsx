import { render, screen } from 'test/test-utils';

import LoadingStatus from './loadingStatus';

describe('<LoadingStatus />', () => {
  test('renders default message', () => {
    render(<LoadingStatus />);
    expect(screen.getByRole('status', { hidden: true })).toBeInTheDocument();
    expect(screen.getByText('No data to show')).toBeInTheDocument();
    expect(screen.getByLabelText('No data')).toBeInTheDocument();
  });

  test('renders custom message', () => {
    const message = 'My message';
    render(<LoadingStatus message={message} />);
    expect(screen.getByText(message)).toBeInTheDocument();
    expect(screen.getByLabelText('No data')).toBeInTheDocument();
  });

  test('renders loading status', () => {
    render(<LoadingStatus isLoading />);
    expect(screen.getByRole('progressbar')).toBeInTheDocument();
    expect(screen.getByText('Loading...')).toBeInTheDocument();
    expect(screen.getByLabelText('Loading')).toBeInTheDocument();
  });

  test('renders error status', () => {
    render(<LoadingStatus isError />);
    expect(screen.queryByRole('progressbar')).not.toBeInTheDocument();
    expect(screen.getByText('No data to show')).toBeInTheDocument();
    expect(screen.getByLabelText('Error')).toBeInTheDocument();
  });
});
