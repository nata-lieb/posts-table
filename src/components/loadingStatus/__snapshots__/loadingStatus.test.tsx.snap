// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoadingStatus /> renders without elevation styles 1`] = `
<div
  aria-label="No data"
  class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 css-1xtxkk1-MuiPaper-root"
  role="status"
>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root MuiSvgIcon-colorDisabled MuiSvgIcon-fontSizeMedium css-2kjbcy-MuiSvgIcon-root"
    data-testid="BlockIcon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zM4 12c0-4.42 3.58-8 8-8 1.85 0 3.55.63 4.9 1.69L5.69 16.9C4.63 15.55 4 13.85 4 12zm8 8c-1.85 0-3.55-.63-4.9-1.69L18.31 7.1C19.37 8.45 20 10.15 20 12c0 4.42-3.58 8-8 8z"
    />
  </svg>
  <p
    class="MuiTypography-root MuiTypography-body1 css-1m8eymn-MuiTypography-root"
  >
    No data to show
  </p>
</div>
`;
